### PR TITLE
Remove legacy Promise APIs from Keycloak JS

### LINKED ISSUE #19388: Remove legacy `.success()` and `.error()`  Promise methods from Keycloak JS
The `.success()` and `.error()` methods on Promises for Keycloak JS have been deprecated for [since 2020](https://github.com/keycloak/keycloak/pull/6665/files#diff-eeb68cbf057af16d2dec09ea923a38332685546740f43d5979d758b39181bd89R73). It's time to remove them so the code can be refactored to standardized Promises internally.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #19388

### CODE REVIEW

**Review by mhajas — CHANGES_REQUESTED** [review]
To me, it makes sense to add this change to release notes.

I see the base test suite was skipped for this PR, we should run at least [this class](https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java) for this. I assume this test method will fail:
https://github.com/keycloak/keycloak/blob/08829835ea51dcf5bc3d240b08c4c4bb1d40c148/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java#L815-L829

**Comment by jonkoops** [comment]
@mhajas updated the test you mentioned, that should also kick off the CI.

> To me, it makes sense to add this change to release notes.

Agreed. Do we have a process in place for that? How can I contribute this?

**Review by mhajas — CHANGES_REQUESTED** [review]
Thank you for the update @jonkoops. I added one more comment to the code.

Release notes are here: 
https://github.com/keycloak/keycloak/tree/main/docs/documentation/release_notes

You need to create a new file `22_0_0.adoc` (interesting it is not there already) and add it to [index.adoc](https://github.com/keycloak/keycloak/blob/main/docs/documentation/release_notes/index.adoc). You can just follow how it is done for [21_0_0.adoc](https://github.com/keycloak/keycloak/blob/main/docs/documentation/release_notes/topics/21_0_0.adoc).

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java [hunk 1]
```diff
@@ -814,9 +814,9 @@ public void fragmentInLoginFunction() {
     @Test
     public void testRefreshTokenWithDeprecatedPromiseHandles() {
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java]: Ahh, yeah agreed. I went ahead and removed this test entirely.

**Comment by jonkoops** [comment]
@mhajas I've removed the test for the legacy methods and added some documentation the docs. Can I ask you to take another look?

File: docs/documentation/release_notes/topics/22_0_0.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
+= Legacy Promise API removed from Keycloak JS
+
+With this release, we have removed the legacy Promise API methods from Keycloak JS. This means that calling `.success()` and `.error()` on promises returned from the adapter is no longer possible. Instead standardized Promise methods such as https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then[`.then()`] and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch[`.catch()`] should be used.
+
+*Before:*
```
Comment (mhajas) [inline docs/documentation/release_notes/topics/22_0_0.adoc:5]: Sorry, I should have told you before. Release notes should contain only a brief description of what was changed. Details on what migration steps are needed should be mentioned in [upgrading docs](https://github.com/keycloak/keycloak/blob/main/docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc).

See, for example, this PR to the old docs repository: https://github.com/keycloak/keycloak-documentation/pull/1760/files

**Review by mhajas — CHANGES_REQUESTED** [review]
Thanks for the update @jonkoops. I added two more comments. 

Also, I just realized you don't have `Closes #19388` in your commit message, can you [add it there as well](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#commit-messages-and-issue-linking)?

**Comment by jonkoops** [comment]
> Also, I just realized you don't have `Closes #19388` in your commit message, can you [add it there as well](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#commit-messages-and-issue-linking)?

This will get added when we merge the PR through GitHub's UI.

File: docs/documentation/release_notes/topics/22_0_0.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
+= Legacy Promise API removed from Keycloak JS
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply docs/documentation/release_notes/topics/22_0_0.adoc]: Changed this everywhere 'Keycloak JS' is mentioned.

**Comment by jonkoops** [comment]
@mhajas I updated the docs with your suggestions, can you take another look?

**Review by mhajas — DISMISSED** [review]
Thank you for the update. The PR looks good to me now.

> > Also, I just realized you don't have `Closes #19388` in your commit message, can you [add it there as well](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#commit-messages-and-issue-linking)?
> 
> This will get added when we merge the PR through GitHub's UI.

The issue number is not added to the commit message on merge.

I am approving this PR because code-wise it looks good to me, however, regarding the commit message I think we should either follow the rules from the [CONTRIBUTING.md](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md) guide or start a discussion for changing it so we can all agree on one process.

**Review by stianst — DISMISSED** [review]
LGTM haven't done a review of the code though, so rely on others to do that.

**Comment by jonkoops** [comment]
> I am approving this PR because code-wise it looks good to me, however, regarding the commit message I think we should either follow the rules from the [CONTRIBUTING.md](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md) guide or start a discussion for changing it so we can all agree on one process.

Yeah, I started a discussion on the mailing list about this but it ended up dying out. My peeve is that issue numbers in commits were a thing when we were still using JIRA. Now that we've got everything on GitHub it will automatically put the PR number in the commit title when merging to `main`, making this procedure less relevant.

**Review by agagancarczyk — APPROVED** [review]

**Review by vmuzikar — COMMENTED** [review]
Does this have any impact on quickstarts?

**Review by pedroigor — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**